tests/TestCase/Authenticator/PrimaryKeySessionAuthenticatorTest.php

```diff
@@ -0,0 +1,441 @@
+
+        $this->identifiers = new IdentifierCollection([
+            'Authentication.Token' => [
+                'tokenField' => 'id',
+                'dataField' => 'key',
+                'resolver' => [
+                    'className' => 'Authentication.Orm',
+                    'userModel' => 'AuthUsers',
+                    'finder' => 'auth',
+                ],
+            ],
+        ]);
+
+        $authenticator = new PrimaryKeySessionAuthenticator($this->identifiers, [
+        ]);
+        $result = $authenticator->authenticate($request);
+
+        $this->assertInstanceOf(Result::class, $result);
+        $this->assertSame(Result::SUCCESS, $result->getStatus());
+
+        $entity = $result->getData();
+        $this->assertNotEmpty($entity->username);
+    }
+
+    /**
+     * Test authentication
+     *
+     * @return void
+     */
+    public function testAuthenticateFailure()
+    {
+        $request = ServerRequestFactory::fromGlobals(['REQUEST_URI' => '/']);
+
+        $this->sessionMock->expects($this->once())
+            ->method('read')
+            ->with('Auth')
+            ->willReturn(null);
+
+        $request = $request->withAttribute('session', $this->sessionMock);
+
+        $authenticator = new PrimaryKeySessionAuthenticator($this->identifiers);
+        $result = $authenticator->authenticate($request);
+
+        $this->assertInstanceOf(Result::class, $result);
+        $this->assertSame(Result::FAILURE_IDENTITY_NOT_FOUND, $result->getStatus());
+    }
+
+    /**
+     * Test session data verification by database lookup failure
+     *
+     * @return void
+     */
+    public function testVerifyByDatabaseFailure()
+    {
+        $request = ServerRequestFactory::fromGlobals(['REQUEST_URI' => '/']);
+
+        $this->sessionMock->expects($this->once())
+            ->method('read')
+            ->with('Auth')
+            ->willReturn(999);
+
+        $request = $request->withAttribute('session', $this->sessionMock);
+
+        $authenticator = new PrimaryKeySessionAuthenticator($this->identifiers, [
+        ]);
+        $result = $authenticator->authenticate($request);
+
+        $this->assertInstanceOf(Result::class, $result);
+        $this->assertSame(Result::FAILURE_IDENTITY_NOT_FOUND, $result->getStatus());
+    }
+
+    /**
+     * testPersistIdentity
+     *
+     * @return void
+     */
+    public function testPersistIdentity()
+    {
+        $request = ServerRequestFactory::fromGlobals(['REQUEST_URI' => '/']);
+        $request = $request->withAttribute('session', $this->sessionMock);
+        $response = new Response();
+        $authenticator = new PrimaryKeySessionAuthenticator($this->identifiers);
+
+        $data = new ArrayObject(['id' => 1]);
+
+        $this->sessionMock
+            ->expects($this->exactly(2))
+            ->method('check')
+            ->with(
+                ...static::withConsecutive(['Auth'], ['Auth']),
```

Comment:
Why `static`?

Reply:
This is copy and paste from the other test.